### DISCUSSION TITLE
React native doctor persistently shows adb and java jdk missing

### DISCUSSION THREAD

**Discussion by JulianHSC** [microsoft/react-native-windows#12915] [Q&A] [answered]
Hello,

I have this problem:
C:\Z-Android\RN_Button>npx react-native doctor
Android
 ✖ Adb - No devices and/or emulators connected. Please create emulator with Android Studio or connect Android device.
 ✓ JDK - Required to compile Java code
 ✓ Android Studio - Required for building and installing your app on Android
 ✓ ANDROID_HOME - Environment variable that points to your Android SDK installation
 ✓ Gradlew - Build tool required for Android builds
 ✖ Android SDK - Required for building and installing your app on Android
   - Versions found: N/A
   - Version supported: 34.0.0
Errors:   2
Warnings: 1

Usage
 › Press f to try to fix issues.
 › Press e to try to fix errors.
 › Press w to try to fix warnings.
 › Press Enter to exit.
I've tried:
-Installing JRE compatible with jfk 17.
-JDK doesn't exist, it's in SDK. I've tried installing JDK 22 with 17 but native doctor will detect 22 and ignore 17. I was forced to uninstall jdk 22.
-Install SDK 34.0, 34.0.0r3, r2, r1 - native doctor still says SDK not detected.
-Inserted C:\Users\(Users)\AppData\Local\Android\Sdk\platform-tools in my system variables.
-Used "f", "e" and "w" to try fixing the problem. It just won't work.

What did i miss out?

**Comment by chrisglein** [comment]
Are you building an Android app or a Windows app? This repo is support for building Windows apps, but sounds like you're trying to do the former. Is that correct?

**Comment by JulianHSC** [comment, marked as answer]
Hello Chris, 
Yes, I'm trying to build an android app, not a Windows app. Your reply is partially too late as i just found out the solution. For unknown reasons, the path is not correct. Once i got the path correct - had to set in command prompt, it worked. You may check the next item i post.

**Reply by chrisglein** [reply]
For future reference, if you have problems building Android apps you should engage with this repo instead: https://github.com/facebook/react-native
